### PR TITLE
Fix error in ctapipe-process in case telescope event is missing true image

### LINKED ISSUE #2344: true image rarely missing for bright events, results in ctapipe-process error
**Please describe the use case that requires this feature.**

It seems that due to a bug or a memory limitation, sometimes the true image data block is completely missing in sim-telarray files for single, especially bright events.

If writing the true images, that results in an exception as the DataWriter tries to write a `None` value into the table setup for writing the true images.

This was reported by @TjarkMiener, observed on 4-LST SiPM and PMT camera simulations, but it seems also observed by Matthieu Heller in the SST-1M processing.

**Describe the solution you'd like**

We should maybe fill these missing true images with -1 or zero values instead of None to allow processing of such files.

A more general solution would be to *optionally* catch exceptions in the event-loop and continue processing, however it is maybe hard to guarantee a well-defined state of the output files then, especially if like here the error happens during a write operation and some values are already written to the file for the failing event.

### PR DESCRIPTION
**Pull request by maxnoe**
Fixes #2344

### CODE REVIEW

**Comment by maxnoe** [comment]
Unfortunately, it's very hard to create a test case for this

**Review by kosack — DISMISSED** [review]

**Review by Tobychev — DISMISSED** [review]

File: src/ctapipe/io/simteleventsource.py [hunk 1]
```diff
@@ -872,6 +873,18 @@ def _generate_events(self):
                     .get("photoelectrons", None)
                 )
 
+                if self._has_true_image is None:
+                    self._has_true_image = true_image is not None
+
+                if self._has_true_image and true_image is None:
+                    self.log.warning(
+                        "Encountered telescope event with missing true_image in"
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply src/ctapipe/io/simteleventsource.py:881]: I'll add a check on the pe sum and o let issue the warning for the "extremely bright case"

**Comment by maxnoe** [comment]
At least for now, we didn't encounter the problematic "event too bright for memory limits" case  in prod6.

Only the non problematic "event very dim". 

I'd process prod6 with this fix and then check the photo electron sums against the images with missing true images to check if we also need to worry about the "too bright case"

**Review by kosack — APPROVED** [review]

File: docs/changes/2659.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Fix error in ``ctapipe-process`` when in the middle of a simtel file
+that has true images available, a telescope event is missing the true image.
+This can happen rarely in case a telescope triggered on pure NSB or
+is oversaturated to the point where the true pe didn't fit into memory constraints.
+
+The error was due to the ``DataWriter`` trying to write a ``None`` into an
+already setup table for the true images.
+
+The ``SimTelEventSource`` will no create an invalid true image filled with ``-1``
```
[the remark this answers is not in the document]
Comment (maxnoe) [reply docs/changes/2659.bugfix.rst]: thanks, fixed.

**Review by Tobychev — APPROVED** [review]

**Comment by maxnoe** [comment]
@kosack @Tobychev with the new test file, sonarqube now passes. Please have a look again

**Review by kosack — APPROVED** [review]

**Pull request merged**